ff/src/fields/models/fp/montgomery_backend.rs

```diff
@@ -241,7 +241,8 @@ pub trait MontConfig<const N: usize>: 'static + Sync + Send + Sized {
             r.b1[i] = carry;
             carry = 0;
         }
-        r.b1[N - 1] >>= 63;
+
+        r.b1[N - 1] = r.b1[N - 2] >> 63;
```

Comment:
```suggestion
        r.b1[N - 1] = r.b1[N - 2] >> 63;
        
```

Reply:
Why is this correct?

Reply:
The code here is indeed to do a "<< 1" across all the limbs. However, the highest limb (which would be zero in this implementation) originally didn't receive the bit from the previous limb. This is not an issue for regular modulus because for those this bit is always zero, but for 64*k-bit modulus this bit could be one.